src/app/OperationalDeviceManager.h

```diff
@@ -71,35 +72,38 @@ class CASESessionManager : public Dnssd::ResolverDelegate
         return CHIP_NO_ERROR;
     }
 
-    virtual ~CASESessionManager() { mDNSResolver.Shutdown(); }
+    virtual ~OperationalDeviceManager() { mDNSResolver.Shutdown(); }
 
     /**
      * Find an existing session for the given node ID, or trigger a new session request.
      * The caller can optionally provide `onConnection` and `onFailure` callback objects. If provided,
      * these will be used to inform the caller about successful or failed connection establishment.
+     *
+     * The user shall call ReleaseDevice to free the device in the failure callback.
```

Comment:
Who 'owns' the `OperationalDeviceProxy` object returned in the callback? Who's responsible for free'ing it during success? During failure?